pilot/cmd/pilot-agent/options/options.go

```diff
@@ -98,6 +98,9 @@ var (
 	enableProxyConfigXdsEnv = env.RegisterBoolVar("PROXY_CONFIG_XDS_AGENT", false,
 		"If set to true, agent retrieves dynamic proxy-config updates via xds channel").Get()
 
+	enableInsecureWasmPluginEnv = env.RegisterBoolVar("INSECURE_WASM_PLUGIN", false,
```

Comment:
I thought we are planning to do a per registry configuration? i.e. it should be a list of registries which are allowlisted for insecure pull.

Reply:
IMHO, if we plan to do `per registry configuration`, `WasmPlugin` seems suitable for this scenario. there's why I put a TODO on `ImageFetcherOption `.

Reply:
I don't think we should ever expose an insecure option in the first class API.

Reply:
add `InsecureRegistries` in `MeshConfig`?

Reply:
Meahconfig is not visible to the agent, and if we 'wire' it via injection will end up as an env variable anyways.

Unless you mean ProxyConfig.

Either way, please qualify it - wasmInsecure... - there are other registries used.

Reply:
> Meahconfig is not visible to the agent, and if we 'wire' it via injection will end up as an env variable anyways.
> 
> Unless you mean ProxyConfig.
> 
> Either way, please qualify it - wasmInsecure... - there are other registries used.

we need some config like `insecure-registries` in `/etc/docker/daemon.json`, @costinm do you have any suggestion?

Reply:
> add InsecureRegistries in MeshConfig?

Adding it to ProxyConfig seems like an overkill to me. If the concern is to configure some registries to be insecure mesh wide, you can already do that by setting ProxyMetadata at mesh wide ProxyConfig.

Reply:
Let's make it a registry whitelist env for now?

Reply:
> Let's make it a registry whitelist env variable for now?

pass array by `ProxyConfig` sounds a bit odd

Reply:
> Let's make it a registry whitelist env variable for now?

How about using `WASM_INSECURE_REGISTRIES`, to disable the insecure feature if it is empty.

Reply:
+1 ProxyConfig is a good place for it.